### PR TITLE
[not WIP] credits3 return of credits [finished] dawn of the night of the credits client preferences [media server updated] return of APC destroyed [does the things] [notice me]

### LINKED ISSUE #20897: credits issues
#20333
 - People with no ID who logged off before roundend are shown as "No ID", should say "as themselves" or use job from crew manifest
 - Sound tends to interrupt after a little while
 - Credits often don't play, neither music nor text, which is very strange 
 - ~~Some people report the sound plays twice~~

### PR DESCRIPTION
**Pull request by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE**
Fixes #20897 fully
I didn't even get to test the last PR so we're doing everything in one go

Super smooth audio playing even for slow connections
Adds client preferences
Adds a shitton of episode names
Adds the thing kurfurst wanted
Adds more logos

If a tator hijacks the shuttle they also hijack the NT logo in the credits

If a clown-related episode name rolls the credits become comic sans in random crayon colors

If 3+ chefs are alive on the shuttle the credits get special styling

:cl:
 * experimental: Updated credits playback method yet again. This should make it smoother than ever, but if you find any problems, please make a bug report.
 + rscadd: Added new client preferences for credits. You can now disable credits if you don't want them, see them every time, or a third option called 'No Reruns' which will show you credits only if an admin customized the credits or if a rare and unusual episode name was automatically chosen because of something uncommon happening that round.
 + rscadd: Added a 'Jingle' client preference. If credits don't play due to your client preferences, then the old round-end 'yeah there we go APC destroyed' jingles will play instead. You can also turn these off completely.
 + rscadd: More possible episode names. Some extra-rare names will also modify the credits themselves.
 + rscadd: Credits casting now shows silicons and doesn't show cloned corpses.
 + rscadd: Admins can now manually set the star of the credits.
 + bugfix: Fixed being able to call the shuttle earlier than 5 minutes after roundstart if the roundstart was delayed.

### CODE REVIEW

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -4834,9 +4834,28 @@
 
 	else if(href_list["credits"])
 		switch(href_list["credits"])
+			if("resetstar")
+				if(!end_credits.drafted) //Just in case the button somehow gets clicked when it shouldn't
+					end_credits.star = null
+					log_admin("[key_name(usr)] reset the current round's featured star. A new one will automatically generate later.")
+					message_admins("[key_name_admin(usr)] reset the current round's featured star. A new one will automatically generate later.")
+			if("setstartext")
+				var/newstar = thebigstar(input(usr,"Write the new star. In the final credits, it will be displayed as: 'Starring\[linebreak\]\[whatever you type here\]'. Mind your capitalization! You may also use HTML. Do not include the characters '%<splashbreak>' anywhere unless you know what you're doing, please.","in my dream, i am the star. its me",null) as text|null)
```
Comment (Kurfursten) [inline code/modules/admin/topic.dm]: it keeps happening

File: __DEFINES/role_datums_defines.dm [hunk 2]
```diff
@@ -46,7 +46,6 @@
 #define MALF "malfunctioning AI"
 #define VOXRAIDER "vox raider"
 #define BLOBOVERMIND "blob overmind"
-#define HIGHLANDER "highlander"
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline __DEFINES/role_datums_defines.dm]: Was duplicate definition

File: code/modules/credits/episode_name.dm [hunk 3]
```diff
@@ -12,82 +16,378 @@
...
+
+			if(human_escapees.len == 1)
+				var/shoecount = 0
+				for(var/obj/item/clothing/shoes/S in shuttle) //they gotta be on the floor
+					shoecount++
+				if(shoecount > 5 || score["shoeshatches"] > 10)
+					episode_names += new /datum/episode_name/rare("THE SOLE SURVIVOR", "There was only one survivor in the shuttle, and they didn't forget their shoes.", 1500) //I'm not sorry
+
+			var/braindamage_total = 0
+			var/all_retarded = TRUE
```
Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [inline code/modules/credits/episode_name.dm]: makes one think

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Okay now I'm done adding shit

**Comment by d3athrow** [comment]
thread told me to merge this so i am

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Absolute madchicken

**Comment by SonixApache** [comment]
waiting for secret repo update general

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
oof ouch owie my secret @Exxion

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
wait @SonixApache this seems to be in the live serb just right

**Comment by SonixApache** [comment]
According to @Kurfursten, the functionality is there, but the changelog hasn't updated.

**Comment by Exxion** [comment]
It looks like MoMMI didn't notice this changelog for some reason

**Pull request merged**